bugsnag-plugin-android-exitinfo/src/androidTest/java/com/bugsnag/android/BugsnagExitInfoPluginStoreTest.kt

```diff
@@ -0,0 +1,119 @@
+package com.bugsnag.android
+
+import android.content.Context
+import androidx.test.core.app.ApplicationProvider
+import com.bugsnag.android.internal.ImmutableConfig
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNull
+import org.junit.Before
+import org.junit.Test
+import java.io.File
+
+internal class BugsnagExitInfoPluginStoreTest {
+
+    private lateinit var file: File
+    private lateinit var exitInfoPluginStore: ExitInfoPluginStore
+    private lateinit var immutableConfig: ImmutableConfig
+
+    @Before
+    fun setUp() {
+        immutableConfig = TestHooks.convertToImmutableConfig(
+            generateConfiguration().apply {
+                persistenceDirectory = ApplicationProvider.getApplicationContext<Context>().cacheDir
+            }
+        )
+        file = File(immutableConfig.persistenceDirectory.value, "bugsnag-exit-reasons")
+        file.delete()
+        exitInfoPluginStore = ExitInfoPluginStore(immutableConfig)
+    }
+
+    /**
+     * Null should be returned for non-existent files
+     */
+    @Test
+    fun readNonExistentFile() {
+        val expectedResult = null to emptySet<ExitInfoKey>()
+        assertEquals(expectedResult, exitInfoPluginStore.load())
+    }
+
+    /**
+     * Null should be returned for empty files
+     */
+    @Test
+    fun readEmptyFile() {
+        file.createNewFile()
+        val expectedResult = null to emptySet<ExitInfoKey>()
+        assertEquals(expectedResult, exitInfoPluginStore.load())
+    }
+
+    /**
+     * Null should be returned for invalid file contents
+     */
+    @Test
+    fun readInvalidFileContents() {
+        file.writeText("{\"hamster\": 2}")
+        val expectedResult = null to emptySet<ExitInfoKey>()
+        assertEquals(expectedResult, exitInfoPluginStore.load())
+    }
+
+    /**
+     * Information should be returned for valid files
+     */
+    @Test
+    fun readValidFileContents() {
+        file.writeText("12345")
+        exitInfoPluginStore = ExitInfoPluginStore(immutableConfig)
+        val info = requireNotNull(exitInfoPluginStore.load())
+        assertEquals(12345, info.first)
+    }
+
+    @Test
+    fun writableFileWithEmptyExitInfo() {
+        exitInfoPluginStore.persist(12345, emptySet())
+        var pid = exitInfoPluginStore.load().first
+        assertNull(pid)
+        exitInfoPluginStore = ExitInfoPluginStore(immutableConfig)
+        pid = exitInfoPluginStore.load().first
+        assertEquals(12345, pid)
+        assertEquals(emptySet<ExitInfoKey>(), exitInfoPluginStore.load().second)
```

Comment:
avoid calling `load()` twice here, we want to assert that both values are correctly returned by the same `load()` call